### DISCUSSION TITLE
Docker image update ignored for a new patch version

### DISCUSSION THREAD

**Discussion by pascalberger** [renovatebot/renovate#16938] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I've a Dockerfile using a base image with a pinned version consisting of Major.Minor.Patch:

```
FROM swissgrc/azure-pipelines-azurecli:2.39.0
```

After I released a new version of the base image with version number `2.39.0.1` I would have expected Renovate to update to the new version, since it is a higher version based on SamVer rules, but Renovate doesn't seem to recognize this.

Can be reproduced on https://github.com/swissgrc/docker-azure-pipelines-helm

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: packageFiles with updates
{
  "config": {
    "dockerfile": [
      {
        "packageFile": "Dockerfile",
        "deps": [
          {
            "depName": "swissgrc/azure-pipelines-azurecli",
            "currentValue": "2.39.0",
            "replaceString": "swissgrc/azure-pipelines-azurecli:2.39.0",
            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "datasource": "docker",
            "depType": "final",
            "depIndex": 0,
            "warnings": [],
            "versioning": "docker",
            "sourceUrl": "https://github.com/swissgrc/docker-azure-pipelines-azurecli",
            "currentVersion": "2.39.0",
            "fixedVersion": "2.39.0",
            "updates": []
          }
        ]
      }
    ],
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by viceice** [comment, marked as answer]
This works by design with default docker versioning[^1], so you need to configure a custom versioning for that image, probably regex[^2]

[^1]: https://docs.renovatebot.com/modules/versioning/#docker-versioning
[^2]: https://docs.renovatebot.com/modules/versioning/#regex

**Reply by pascalberger** [reply]
Thanks for the explanation 👍

**Reply by rarkins** [reply]
Or loose versioning?

**Reply by pascalberger** [reply]
Yes, loose versioning (or semantic versioning) seems to be what I expect for this specific image

**Reply by rarkins** [reply]
What you described above is not semantic versioning compliant
